### ISSUE TITLE
paddle.fluid.layers.cross_entropy(input, label, soft_label=True) 运行后label被更改

### ISSUE THREAD

**Issue by linrjing** [PaddlePaddle/models#4410]
loss 使用
fluid.layers.cross_entropy(input=self.network_outputs[0], label=gt_label, soft_label=True)，

输入：
gt_label是我自己设定好的  [0.  0.  0.  0.5 0.  0.5 0.  0.  0.  0.  0.  0.  0.  0.  0.  0. ]

在train的run之后打印 fetch_out的gt_label变为：
![F2D45D6252E53F5B661495BB63587376](https://user-images.githubusercontent.com/9814501/76505363-556eb200-6484-11ea-839b-393f72dab3a8.JPG)

loss完整代码如下：
        cost_cls = fluid.layers.cross_entropy(input=self.network_outputs[0], label=self.label_id_input, soft_label=True)
        #cost_cls = fluid.layers.sigmoid_cross_entropy_with_logits(x=self.logit, label=self.label_id_input)
        cost_cls = fluid.layers.reduce_sum(cost_cls, dim=-1)
        sum_cost_cls = fluid.layers.reduce_sum(cost_cls)
        self.loss_cls_ = fluid.layers.scale(sum_cost_cls, scale=self.num_gpus, bias_after_scale=False)

尝试用 fluid.layers.sigmoid_cross_entropy_with_logits.  gt_label不变化

**Comment by Aurelius84** [comment]
我使用如下代码，并未复现问题：
```python
import paddle.fluid as fluid
import numpy as np

x = fluid.layers.data(shape=[-1, 3], name='x', dtype='float32')
y = fluid.layers.data(shape=[-1, 3], name='y', dtype='float32')
y = fluid.layers.reshape(y, [-1, 3], inplace=True)

pre = fluid.layers.fc(x, size=3, act='softmax')
cost = fluid.layers.cross_entropy(input=pre, label=y, soft_label=True)
loss = fluid.layers.mean(cost)

sgd = fluid.optimizer.SGD(learning_rate=0.01)
sgd.minimize(loss)

exe = fluid.Executor(fluid.CPUPlace())
exe.run(fluid.default_startup_program())
for i in range(3):
    x_data = np.random.random([1, 3]).astype('float32')
    y_data = np.array([[0, 0.5, 0.5]]).astype('float32')
    out = exe.run(feed={'x': x_data, 'y': y_data}, fetch_list=y)
    print(out)
```

**Comment by linrjing** [comment]
bug出现了fluid1.5.2。 重新安装fluid1.6后， luid.layers.cross_entropy(input=pre, label=y, soft_label=True)正常了
